src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -52,24 +52,22 @@ public override async Task<T> GetAssetAsync<T>(Checksum checksum, CancellationTo
             }
         }
 
-        public async Task<List<ValueTuple<Checksum, T>>> GetAssetsAsync<T>(IEnumerable<Checksum> checksums, CancellationToken cancellationToken)
+        public async ValueTask<ImmutableArray<ValueTuple<Checksum, T>>> GetAssetsAsync<T>(HashSet<Checksum> checksums, CancellationToken cancellationToken)
```

Comment:
passing items with known length allows us to optimize more (both by bailing out early and by getting the right sized immutable collections to return).

Reply:
Should we use an Immutable type here?